File: BehaviorPattern/Chain_Of_Responsibility/Java/Concretes/Level1SupportHandler.java
package Concretes;

import Interfaces.SupportHandler;

public class Level1SupportHandler extends SupportHandler {
    @Override
    public void handleRequest(SupportRequest request) {
        if (request.getComplexity() <= 1) {
            System.out.println("Level 1 support handled the request: " + request.getMessage());
        } else if (nextHandler != null) {
            nextHandler.handleRequest(request);
        }
    }
}

File: BehaviorPattern/Chain_Of_Responsibility/Golang/Concretes/Level1SupportHandler.go
package concretes

import (
    "fmt"
    "v"
    "v/Interfaces"
)

type Level1SupportHandler struct {
    nextHandler interfaces.SupportHandler
}

func (h *Level1SupportHandler) SetNext(handler interfaces.SupportHandler) {
    h.nextHandler = handler
}

func (h *Level1SupportHandler) HandleRequest(request *v.SupportRequest) {
    if request.GetComplexity() <= 1 {
        fmt.Println("Level 1 support handled the request:", request.GetMessage())
    } else if h.nextHandler != nil {
        h.nextHandler.HandleRequest(request)
    }
}
